keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -833,6 +836,75 @@ public DoubleTensor concat(int dimension, DoubleTensor... those) {
         return new Nd4jDoubleTensor(concat);
     }
 
+    /**
+     * @param dimension      the dimension to split on
+     * @param splitAtIndices the indices that the dimension to split on should be split on
+     * @return pieces of the tensor split in the order specified by splitAtIndices. To get
+     * pieces that encompasses the entire tensor, the last index in the splitAtIndices must
+     * be the length of the dimension being split on.
+     * <p>
+     * e.g A =
+     * [
+     * 1, 2, 3, 4, 5, 6
+     * 7, 8, 9, 1, 2, 3
+     * ]
+     * <p>
+     * A.split(0, [1]) gives List([1, 2, 3, 4, 5, 6])
+     * A.split(0, [1, 2] gives List([1, 2, 3, 4, 5, 6], [7, 8, 9, 1, 2, 3]
+     * <p>
+     * A.split(1, [1, 3, 6]) gives
+     * List(
+     * [1, [2, 3  , [4, 5, 6,
+     * 7]   8, 9]    1, 2, 3]
+     * )
+     */
+    @Override
+    public List<DoubleTensor> split(int dimension, int[] splitAtIndices) {
+
+        DoubleTensor duplicateTensor = this.duplicate();
+        int[] tensorShape = duplicateTensor.getShape();
+        int[] dimensionRange = TensorShape.dimensionRange(0, tensorShape.length);
+        int[] moveDimToZero = TensorShape.moveAxis(dimension, 0, dimensionRange);
+        int[] moveZeroToDim = TensorShape.moveAxis(0, dimension, dimensionRange);
+
+        DoubleTensor permutedTensor = duplicateTensor.permute(moveDimToZero).reshape(1, (int) duplicateTensor.getLength());
```

Comment:
Would be cool to pull that flatten out into a function:

```
public DoubleTensor flatten() {
        return this.duplicate().reshape(1, (int) getLength());
    }
```

Reply:
It's only used here and I feel like there are some semantics to define around flatten. Like flatten to column OR row OR to a vector (rank 1). With that in mind I'd say do it in another PR if we find it useful somewhere else.